Added options for :ignore_acknowledgement & :expect_transaction_error

coreyeng opened the pull request:
Had a need to verify that a transaction was failing. Added an option for that and threw a option in to just ignore the acknowledgement entirely, just in case.

redxeth commented on lib/origen_swd/driver.rb:
```diff
@@ -177,11 +177,28 @@ def send_header(apndp, rnw, address)
+    #   transaction, verifies that a transaction error occurred.
+    # @note If both :ignore_acknowledgement and :expect_transaction_error are present,
+    #   :ignore_acknowledgement will take precedence.
+    def receive_acknowledgement(options = {})
```
Should you also have default settings for the various options here?  I know you check all cases below but a good practice.

coreyeng replied:
The default will be `nil`, so they'll all be skipped by `Hash` access definition; essentially `false`. And the defaults would only be used internally, so I'm not sure we gain anything.

coreyeng replied:
Unless you're talking about the YARDOC specifically?

coreyeng commented:
I changed it around to use a keyword argument instead. Should be a bit clearer. @redxeth, what do you think of this?

I also, somewhat embarrassingly, kept misspelling acknowledgment. Not a word I need to type out often. Anyway, I changed the option name to `confirm`, as in `confirm acknowledgment ...`. May make it a bit more user friendly, and could have applicability for drivers that support it without being too SWD specific.
